### PR TITLE
Updated the documentation.

### PR DESCRIPTION
**Pull request by Victorsitou**
## Summary

Preview can be found [here](https://pr-151-preview.readthedocs.io/en/latest/).

I have reviewed the whole documentation and found a few things. (I did my best to change it properly)

I have some questions about the documentation.

- Should we refer to `GuildScheduledEvent` as `guild scheduled event` or just `scheduled event`? (Some descriptions have "`?`", those are related to this question, as well as the other descriptions that refer to this)

- It looks like raw events aren't implemented, do you plan to implement them? If not, then let me know so I can update the intents section.

Also, I made a poll in the Discord server asking if `on_scheduled_event_create/delete` should be listed together, I would like to hear your opinion about it. [Link to the message](https://discord.com/channels/808030843078836254/911640773580185652/911787084338053141).

As for now, I listed them together and left them alone, so that is a planned change to make.

Finally, I really want to thanks to the people who have helped me :)

## Checklist

- [ ] If code changes were made then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `black .`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by Victorsitou** [comment]
Also, in line [1144](https://github.com/EQUENOS/disnake-1/pull/2/files#diff-0ee295c29def16f570f86378eff8ba37315e464d415e731d5d8ea2e67ce16f68R1144-R1145) and [1157](https://github.com/EQUENOS/disnake-1/pull/2/files#diff-0ee295c29def16f570f86378eff8ba37315e464d415e731d5d8ea2e67ce16f68R1157-R1158) of `api.rst`, should those parameters be type ``Union[:class:`Member`, :class:`User`]``? 
Since the source-code [supports both](https://github.com/DisnakeDev/disnake/pull/151/files#diff-3576d5b144cbf52422d024d3aa0d6353a3d44548b6c691226c97df1024d9f440R1486-R1488).

Also, if we're going to list `on_scheduled_event_create/delete` together, then we should list together those events as well.

File: disnake/guild_scheduled_event.py [hunk 1]
```diff
@@ -109,20 +109,20 @@ class GuildScheduledEvent(Hashable):
...
     description: :class:`str`
         The description of the scheduled event (1-1000 characters).
-    scheduled_start_time: :class:`datetime`
-        The time the event will start.
-    scheduled_end_time: Optional[:class:`datetime`]
-        The time the event will end, or ``None`` if the event does not have a scheduled time to end.
+    scheduled_start_time: :class:`datetime.datetime`
+        The time when the scheduled event is scheduled to start.
+    scheduled_end_time: Optional[:class:`datetime.datetime`]
+        The time when the scheduled event will end, or ``None`` if the event does not have a scheduled time to end.
```
Comment (Victorsitou) [inline disnake/guild_scheduled_event.py:115]: Should be `scheduled_end_time` description the same as `start_time`? I mean, follow basically the same syntax, no?

Comment (EQUENOS) [reply disnake/guild_scheduled_event.py:115]: "The time when the scheduled event will start" sounds better imho, so yeah

**Comment by EQUENOS** [comment]
Turns out that I didn't pay enough attention to the event names in the docs when I restored the documentation from my previous implementation of scheduled events. The event names are actually `on_guild_scheduled_event_xxx`, not `on_scheduled_event_xxx`

**Comment by Victorsitou** [comment]
> Turns out that I didn't pay enough attention to the event names in the docs when I restored the documentation from my previous implementation of scheduled events. The event names are actually `on_guild_scheduled_event_xxx`, not `on_scheduled_event_xxx`

Oh, then are you going to update the events names?
 
> Also I suggest referencing the scheduled events using "guild scheduled event" term for consistency.

Alright, I'm going to update the documentation when I have time, thank you.

**Comment by Victorsitou** [comment]
Also, it seems that shiftinv made some changes, I'm gonna merge them here.

**Comment by shiftinv** [comment]
Didn't know this PR existed, I already pushed some tiny documentation fixes here, sorry: https://github.com/DisnakeDev/disnake/pull/151/commits/e8d7ee1ef9d35745443946ff099a8fc8937183be
I'll could that commit again to avoid merge conflicts, since those changes are already contained in this PR for the most part.

Also, I'd suggest using `Retrieves ...` instead of `Returns ...` for the `fetch_*` methods, in line with similar methods in the `Guild` class.

**Comment by Victorsitou** [comment]
> Didn't know this PR existed, I already pushed some tiny documentation fixes here, sorry: [DisnakeDev@e8d7ee1](https://github.com/DisnakeDev/disnake/commit/e8d7ee1ef9d35745443946ff099a8fc8937183be) I'll could revert that commit again to avoid merge conflicts, since those changes are already contained in this PR for the most part.

It's ok shiftinv, no worries! And yeah, if you could revert those changes I would appreciate it (Basically because I don't want to mess up the PR by doing unnecessary stuff in the git bash haha)

**Comment by shiftinv** [comment]
Done 👍

File: disnake/guild_scheduled_event.py [hunk 2]
```diff
@@ -208,12 +208,12 @@ def __str__(self) -> str:
...
     @cached_slot_property("_cs_guild")
     def guild(self) -> Optional[Guild]:
-        """:class:`Guild` The guild which the scheduled event belongs to."""
+        """:class:`Guild` The guild which the scheduled event belongs to. what is this?"""
         return self._state._get_guild(self.guild_id)
 
     @cached_slot_property("_cs_channel")
     def channel(self) -> Optional[GuildChannel]:
-        """:class:`GuildChannel` The channel in which the scheduled event will be hosted."""
+        """:class:`GuildChannel` The channel in which the scheduled event will be hosted. what is this?"""
```
Comment (Victorsitou) [inline disnake/guild_scheduled_event.py:216]: What are those properties and why are these docstrings outside the class docstrings?

image

File: disnake/guild.py [hunk 3]
```diff
@@ -1801,7 +1801,7 @@ async def fetch_scheduled_events(
         Returns
         --------
         List[:class:`GuildScheduledEvent`]
-            The scheduled events.
+            The guild scheduled events. (A list of all the existing guild scheduled events. | The existing guild scheduled events.)
```
Comment (Victorsitou) [inline disnake/guild.py:1804]: What should be the description of the return? I added some candidates there.

File: docs/api.rst [hunk 4]
```diff
@@ -2923,25 +2915,25 @@ of :class:`enum.Enum`.
 
 .. class:: GuildScheduledEventEntityType
 
-    Represents the type of a scheduled event entity.
+    Represents the type of a guild scheduled event entity. **Or instance?**
```
Comment (Victorsitou) [inline docs/api.rst:2918]: Should it be entity or instance?

**Comment by Victorsitou** [comment]
Ok, so the PR should be ready for review (except for those questions I asked above).

Any suggestions are totally appreciated.

File: disnake/guild.py (hunk 3 above)
Comment (EQUENOS) [reply disnake/guild.py:1804]: The existing guild scheduled events

File: disnake/guild_scheduled_event.py [hunk 5]
```diff
@@ -208,12 +208,19 @@ def __str__(self) -> str:
 
     @cached_slot_property("_cs_guild")
     def guild(self) -> Optional[Guild]:
-        """:class:`Guild` The guild which the scheduled event belongs to."""
+        """The guild which the guild scheduled event belongs to. what is this?
```
Comment (EQUENOS) [inline disnake/guild_scheduled_event.py:211]: Remove "what is this" please

File: disnake/guild_scheduled_event.py [hunk 6]
```diff
@@ -208,12 +208,19 @@ def __str__(self) -> str:
...
+        """
         return self._state._get_guild(self.guild_id)
 
     @cached_slot_property("_cs_channel")
     def channel(self) -> Optional[GuildChannel]:
-        """:class:`GuildChannel` The channel in which the scheduled event will be hosted."""
+        """The channel in which the guild scheduled event will be hosted.
+
+        :return type: Optional[:class:`abc.GuildChannel`]
+
```
Comment (EQUENOS) [inline disnake/guild_scheduled_event.py:222]: Remove the extra line

File: docs/api.rst (hunk 4 above)
Comment (EQUENOS) [reply docs/api.rst:2918]: Entity

**Review by EQUENOS — APPROVED** [review]

**Pull request merged**